remove-webhook/remove-webhook.sh

```diff
@@ -1,24 +1,84 @@
+Requires kubectl and helm to be installed and available on \$PATH.
+
+RANCHER_URL without a trailing slash must be set with the server URL of Rancher.
+RANCHER_TOKEN must be set with an admin token generated with no scope.
+To ignore TLS verification, set --insecure-skip-tls-verify."
+
+This script supports Rancher v2.7.x.
+}
+
+if [ "$1" == "-h" ]; then
+  usage
+  exit 0
+fi
 
 if [ -n "$DEBUG" ]
 then
     set -x
 fi
 
-usage() {
-    echo "./remove-webhook.sh [--insecure-skip-tls-verify]"
-    echo "Remove the webhook chart in all clusters managed by rancher (excluding the local cluster)"
-    echo "Requires kubectl and helm to be installed and available on \$PATH"
-    echo "--insecure-skip-tls-verify can be set to configure the script to ignore tls verification"
-    echo "RANCHER_TOKEN must be set with an admin token generated with no scope"
-    echo "RANCHER_URL must be set with the url of rancher (no trailing /) - should be the server URL"
+delete_webhook() {
+    cluster="$1"
+    current_chart=$(helm list -n cattle-system -l name=rancher-webhook | tail -1 | cut -f 6)
+    echo "Deleting $current_chart from cluster $cluster."
+    kubectl config set clusters.local.server "$RANCHER_URL/k8s/clusters/$cluster"
+    helm uninstall rancher-webhook -n cattle-system
+}
+
+replace_webhook() {
+    cluster="$1"
+    new_version="$2"
+
+    echo "Updating the agent to make it remember the min version $new_version of rancher-webhook, so that it can deploy it when needed in the future in cluster $cluster."
+    kubectl config set clusters.local.server "$RANCHER_URL/k8s/clusters/$cluster"
+    kubectl set env -n cattle-system deployment/cattle-cluster-agent CATTLE_RANCHER_WEBHOOK_MIN_VERSION="$new_version"
+
+    helm get values -n cattle-system rancher-webhook > current_values.yaml
+    delete_webhook "$cluster"
+    echo "Re-installing rancher-webhook to use $new_version in cluster $cluster."
+    helm upgrade --install rancher-webhook rancher-charts/rancher-webhook -n cattle-system --version "$new_version" --values current_values.yaml
+    rm -f current_values.yaml
+}
+
+adjust_webhook() {
+    cluster="$1"
+    rancher_version="$2"
+
+    if [[ "$rancher_version" =~ 2\.7\.[0-1] ]]; then
+        delete_webhook "$cluster"
+    elif [[ "$rancher_version" =~ 2\.7\.2 ]]; then
+        replace_webhook "$cluster" 2.0.2+up0.3.2
+    elif [[ "$rancher_version" =~ 2\.7\.3 ]]; then
+        replace_webhook "$cluster" 2.0.3+up0.3.3
+    elif [[ "$rancher_version" =~ 2\.7\.4 ]]; then
+        replace_webhook "$cluster" 2.0.4+up0.3.4
+    else
+      kubectl config set clusters.local.server "$RANCHER_URL/k8s/clusters/$cluster"
+      kubectl set env -n cattle-system deployment/cattle-cluster-agent CATTLE_RANCHER_WEBHOOK_MIN_VERSION=''
+        echo "No need to delete rancher-webhook, given Rancher version $rancher_version."
+    fi
+}
+
+validate_rancher_version() {
+    version="$1"
+    if [[ ! $version =~ 2\.7\.[0-9] ]]; then
```

Comment:
I don't think that we can limit this to 2.7 versions. I think we also need to consider the case where a user is rolling back from 2.7 to 2.6. This means we will need to have a case above for 2.6.13 (which installs that version of the downstream webhook - [1.0.9+up0.2.10](https://github.com/rancher/rancher/blob/e91bb053386b32864bbbb306f39061e9f2e287ae/package/Dockerfile#L58), and for versions below 2.6.13 (which just remove the webhook).

Reply:
Good point, I will not restrict it to v2.7.x and add a case for v2.6.13. Is this the only 2.6 release that has webhook downstream?

Reply:
Yes I believe so.